utils/keys_utils.go

```diff
@@ -0,0 +1,166 @@
+package utils
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/pem"
+	"io/ioutil"
+	"log"
+)
+
+func Gen(id string) {
+	keyspath := GetPrjPath() + "worker/keys"
+	savePrivateFileTo := keyspath + "/" + id + ".priv"
+	savePublicFileTo := keyspath + "/" + id + ".pub"
+	bitSize := 4096
+
+	privateKey, err := GeneratePrivateKey(bitSize)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	publicKeyBytes := generatePublicKey(&privateKey.PublicKey)
+
+	privateKeyBytes := encodePrivateKeyToPEM(privateKey)
+
+	err = writeKeyToFile(privateKeyBytes, savePrivateFileTo)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	err = writeKeyToFile([]byte(publicKeyBytes), savePublicFileTo)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+}
+
+func GetPrivateKey(id string) *rsa.PrivateKey {
+	readPrivKey, err := ioutil.ReadFile(GetPrjPath() + "worker/keys/" + id + ".priv")
+	if err != nil {
+		log.Fatal("The private key is not where it should be")
+	}
+
+	pemDecodedPrivKey, _ := pem.Decode(readPrivKey)
+
+	rsaPrivateKey, err := x509.ParsePKCS1PrivateKey(pemDecodedPrivKey.Bytes)
+	if err != nil {
+		log.Fatal("Error on parsing private key " + err.Error())
+	}
+
+	return rsaPrivateKey
+}
+
+func SignMessage(privateKey *rsa.PrivateKey, message []byte) ([]byte, []byte) {
+	messageHash := sha256.New()
+	_, err := messageHash.Write(message)
```

Comment:
The Write function returns the number of written bytes, you can use it to double-check if the message was entirely written.